pkg/initializer_v2/test/e2e/test_dataset.py

```diff
@@ -0,0 +1,107 @@
+import os
+import runpy
+import shutil
+import tempfile
+
+import pytest
+
+import pkg.initializer_v2.utils.utils as utils
+from sdk.python.kubeflow.storage_initializer.constants import VOLUME_PATH_DATASET
+
+
+class TestDatasetE2E:
+    """E2E tests for dataset initialization"""
+
+    @pytest.fixture(autouse=True)
+    def setup_teardown(self, monkeypatch):
+        """Setup and teardown for each test"""
+        # Create temporary directory for dataset downloads
+        current_dir = os.path.dirname(os.path.abspath(__file__))
+        self.temp_dir = tempfile.mkdtemp(dir=current_dir)
+        os.environ[VOLUME_PATH_DATASET] = self.temp_dir
```

Comment:
I currently test downloading with creating a temp folder and removing the temp folder after the test. I am not sure if this is OK.